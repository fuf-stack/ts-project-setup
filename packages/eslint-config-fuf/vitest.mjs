// see: https://github.com/veritem/eslint-plugin-vitest

import vitest from '@vitest/eslint-plugin';

export default [
  {
    name: 'fuf/vitest',
    plugins: { vitest },

    // configure file patterns for tests
    files: ['**/*.spec.ts', '**/*.spec.tsx', '**/*.test.ts', '**/*.test.tsx'],

    // test file rules
    rules: {
      // recommended vitest rules
      ...vitest.configs.recommended.rules,

      // Turn off legacy import plugin rules if present upstream
      'import-x/extensions': 'off',
      'import/extensions': 'off',

      // Do not enforce production dependencies in tests
      'import/no-extraneous-dependencies': 'off',
      'import-x/no-extraneous-dependencies': 'off',
    },
  },
];
