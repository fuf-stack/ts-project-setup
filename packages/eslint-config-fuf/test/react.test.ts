import { expect, it } from 'vitest';

import { lintFixture } from './helper';

it('enforces component arrow-function definition', async () => {
  const { fixedContent } = await lintFixture(
    'react-function-component-definition.tsx',
    'react.config.mjs',
  );
  expect(fixedContent).toMatchSnapshot();
});

it('shows warning when optional props do not have a function default arg', async () => {
  const { results } = await lintFixture(
    'react-function-component-default-props.tsx',
    'react.config.mjs',
  );
  expect(results).toMatchObject([
    {
      messages: [
        {
          ruleId: 'react/require-default-props',
          messageId: 'shouldAssignObjectDefault',
          message:
            'propType "foo" is not required, but has no corresponding default argument value.',
        },
      ],
    },
  ]);
});

it('fixes storybook stories', async () => {
  const { fixedContent } = await lintFixture(
    'react-Button.stories.tsx',
    'react.config.mjs',
  );
  expect(fixedContent).toMatchSnapshot();
});
