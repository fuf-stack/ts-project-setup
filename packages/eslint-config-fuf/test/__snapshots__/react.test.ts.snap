// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`enforces component arrow-function definition 1`] = `
"const Component = () => <div />;

export default Component;
"
`;

exports[`fixes storybook stories 1`] = `
"// TODO: currently we only check that formatting is working at all

import type { Meta, StoryObj } from '@storybook/react-vite';

import { Button } from './Button';

const meta: Meta<typeof Button> = {
  component: Button,
};

export default meta;
type Story = StoryObj<typeof meta>;

export const Primary: Story = {
  args: {
    primary: true,
    label: 'Button',
  },
};
"
`;
